### PR TITLE
Moved Profiles just below Info

### PR DESCRIPTION
**Pull request by RicardoEPRodrigues**
Did a little refactor and separated the pop up building function into smaller functions. I think it makes it easier to move the sections up and down.

I wanted to put the profiles on top and it is easier this way.

Here's how it looks! :smile: 

screenshot from 2017-05-21 17-36-28

### CODE REVIEW

**Comment by konkor** [comment]
OMG @RicardoEPRodrigues you have reworked completely all build-popup function... I didn't tell you it's bad. Bad need a lot testing and reviewing. I have used to all in one and see variable initialization )
About ur Profiles moving to the top. I like it idea too but:
1. It's looking like we have to provide some predefined Profiles. It's a some extra code and not respectful work because most users want to make own very special profiles.
2. Profiles didn't have a wide testing yet. I mean not published on gnome.org and they look like have some issues (like you).
3. People don't know how to use it yet.

We can keep this in this new info branch after my review and basic tests... Do you really like this idea with this separate functions or it was for ur learning purposes? :)

**Comment by RicardoEPRodrigues** [comment]
The Pull requests do allow for a lot of testing to be done easily. So I invite anyone to test this out.

I really like the idea of separate functions, mainly because it allows to easily move things around and also increases code readability. I think this division is not perfect yet and I would prefer something like what you have for the info:

    let info = new InfoItem ();
    this.menu.addMenuItem (info);

For now I wanted to know your opinion on doing this. Maybe with this we could have a dynamic popup, where the user decides the order and visibility of the topics.

**Comment by konkor** [comment]
OK maybe you right. Actually I did some encapsulation too but dropped it because it needs time and here is always something to do. Maybe it's time... I will test it and merge to info...
I don't have a special opinions to separate popup on blocks. This menu was just much smaller at beginning. ;) So I'm editing in an text editor (mostly GEdit or Atom) where the moving through the many classes functions is not very easy. So many classes and function calls needs more resources especially on high level languages like JavaScript.
PS: I want to update profiles tree and add full timeouts everywhere on loading profiles. You have to test it so. I'll explain it later.

**Comment by RicardoEPRodrigues** [comment]
Is it possible to use JS uglify and minify to improve performance?

**Comment by konkor** [comment]
I think no. Because moderators have to review code too...

**Comment by RicardoEPRodrigues** [comment]
I may be over-thinking it, but we could create a `compilation` to remove functions and put them inline :smile:

**Comment by konkor** [comment]
Relax. It's not so critical part of code. It's running once on startup only.

File: extension.js [hunk 1]
```diff
@@ -296,299 +296,320 @@ const FrequencyIndicator = new Lang.Class({
...
+                }
+            }));
+        }
+    },
+
+    _build_popup: function () {
+        this.menu.removeAll ();
+        if (this.util_present) {
+            
+
```
Comment (konkor) [inline extension.js]: remove this 2 empty string...

File: extension.js [hunk 2]
```diff
@@ -296,299 +296,320 @@ const FrequencyIndicator = new Lang.Class({
...
+            this._build_profiles();
+            this.menu.addMenuItem (new SeparatorItem ());
+            this._build_governors();
+            this.menu.addMenuItem (new SeparatorItem ());
+            this._build_frequencies();
+            this._build_cpu_power();
+            this._build_turbo_boost();
+            this.menu.addMenuItem (new SeparatorItem ());
+            this._build_preferences();
+
```
Comment (konkor) [inline extension.js]: another one empty string

File: extension.js [hunk 3]
```diff
@@ -296,299 +296,320 @@ const FrequencyIndicator = new Lang.Class({
...
+        this.boost_switch = null;
+        let slider_lock = false;
+        if (this.pstate_present) {
+            this.slider_min = new Slider.Slider (this._get_min_pstate () / 100);
+            this.slider_max = new Slider.Slider (this._get_max_pstate () / 100);
+        } else if (this.frequences.length > 1) {
+            this.slider_min = new Slider.Slider (this._get_pos (this._get_min ()));
+            this.slider_max = new Slider.Slider (this._get_pos (this._get_max ()));
+        }
+
```
Comment (konkor) [inline extension.js]: remove this string with spaces...

**Pull request closed without merging**